### DISCUSSION TITLE
What is the relevance of this copy?

### DISCUSSION THREAD

**Discussion by 1r00t** [codingjoe/django-pictures#70] [Q&A] [answered]
https://github.com/codingjoe/django-pictures/blob/acb0a0df59d746a587dc94a0c5b1606a93259f54/pictures/models.py#L75

What is the relevance of this copy?
Maybe the fix for image rotation (#68) would make this line irrelevant?

**Comment by codingjoe** [comment, marked as answer]
Hi @1r00t,

Excellent question. This is to avoid state leakage. If you iterate over all sizes and don't create a copy before applying any changes to the object, the state will leak to the next iteration of the loop.

I believe this line was part of a bug fix from @vchrisb who discovered this issue.

Cheers,
Joe

**Reply by 1r00t** [reply]
Perfect, so the `exif_transpose` function could replace this copy because it will itself create a copy. From the ImageOps reference:

> If an image has an EXIF Orientation tag, other than 1, return a new image that is transposed accordingly. The new image will have the orientation data removed.

> Otherwise, return a copy of the image.

It still needs to be tested
